### PR TITLE
DDF-04262 Add the ability to set a default list

### PR DESCRIPTION
**Pull request by gordocanchola**
Setting the ID of the List to user preferences for the List chosen as default
Adding the 'Make Default List' option to the 3 dot menu of each List
Adding the 'Clear Default List' option to the 3 dot menu of the List set as default
Adding a comparator to the list-item.collection.view to set default List to top of list

#### What does this PR do?
This PR adds the ability to set a list as default. The default list will move to the top of the list and have the star icon.

#### Who is reviewing it? 
@Schachte @middlets719 @zta6 @adimka @djblue @bdeining 
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@djblue

#### How should this be tested?

Install DDF
Open Intrigue.
Open a workspace
Select the Lists tab
Create a few new lists
Verify they are ordered by name
Open the 3 dot menu of a List and select `Make Default List`
Verify the List is moved to the top of the lists and it has the star icon
Open the 3 dot menu of the default list and select `Clear Default List`
Verify the list is ordered by name again and the there is no longer the star icon

#### What are the relevant tickets?
For GH Issues:
Fixes: #4262 

#### Screenshots
image
`ccccc` has been selected as default in the image above.
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-interactions/list-interactions.view.js [hunk 1]
```diff
@@ -32,13 +33,21 @@ module.exports = Marionette.ItemView.extend({
...
+    'click .interaction-clear': 'triggerClearDefault',
     click: 'triggerClick',
   },
   modelEvents: {
     'change:actions': 'render',
   },
   onRender() {
     this.handleResult()
+    this.handleDefault()
+    this.listenTo(
```
[the remark this answers is not in the document]
Comment (gordocanchola) [reply ui/packages/catalog-ui-search/src/main/webapp/component/list-interactions/list-interactions.view.js]: Changed to initialize()

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-item/list-item.view.js [hunk 2]
```diff
@@ -92,6 +93,12 @@ module.exports = Marionette.LayoutView.extend({
   },
   onRender: function() {
     this.setupFeed()
+    this.handleDefault()
+    this.listenTo(
```
[the remark this answers is not in the document]
Comment (gordocanchola) [reply ui/packages/catalog-ui-search/src/main/webapp/component/list-item/list-item.view.js]: Moved to initialize()

**Comment by Schachte** [comment]
build now

**Comment by gordocanchola** [comment]
build now

**Comment by ethantmanns** [comment]
Manually verified that this works as intended! ✅

**Pull request merged**